src/utils/format.ts

```diff
@@ -98,6 +100,30 @@ export function parseAmount(amountFmt: string, amountPrecision = DEFAULT_PRECISI
   }
 }
 
+/**
+ * Adjust the decimal precision of the given decimal value, without converting to/from BN or Number
+ * Takes in a string and returns a string
+ *
+ * E.g.:
+ * adjustPrecision('1.2657', 3) === '1.265'
+ *
+ * @param value The decimal value to be adjusted as a string
+ * @param precision How many decimals should be kept
+ */
+export function adjustPrecision(value: string | undefined | null, precision: number): string {
+  if (!value) {
+    return ''
+  }
+
+  const match = value.match(/(^\d+\.)(\d+)$/)
+  if (match && match[2].length > precision) {
```

Comment:
It looks very similar to me to `parseAmount`

Wouldn't be worth to split `parseAmount` in two. It would be similar to `_decomposeBn` but accepting a string instead of a BN. One idea would be to overload the method and provide the two signatures